File: CareConnect/CareConnect.Models/Requests/ChildInsertRequest.cs
using System;
using System.Text.Json.Serialization;

namespace CareConnect.Models.Requests
{
    public class ChildInsertRequest
    {
        public string FirstName { get; set; } = null!;

        public string LastName { get; set; } = null!;

        public DateTime BirthDate { get; set; }

        public string Gender { get; set; } = null!;

        [JsonIgnore]
        public DateTime ModifiedDate { get; set; } = DateTime.Now;
    }   
}

File: UI/careconnect_mobile/lib/models/requests/child_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'child_insert_request.g.dart'; // Don't open or edit this

@JsonSerializable()
class ChildInsertRequest {
  String firstName;
  String lastName;
  DateTime birthDate;
  String gender;

  ChildInsertRequest({
    required this.firstName,
    required this.lastName,
    required this.birthDate,
    required this.gender,
  });

  factory ChildInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$ChildInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$ChildInsertRequestToJson(this);
}
